docs/basics/networkvariable.md

```diff
@@ -331,6 +331,123 @@ The above example provides you with details on:
+
+  - **Server/Host Smoothing:** 
+
+    - *Example: The host player is watching another player play the fishing minigame. The player is pressing the button regularly, but due to network latency and jitter, the inputs are not arriving smoothly on the host, resulting in jerky movement of the fish. When new inputs arrive, the host uses smoothing to hide the unevenness of the inputs.*
+    
+    Smoothing can also be performed on the server/host side. To do this, when you receive an RPC from the user to change a value, you can call `variable.Smooth()` from inside the RPC to smooth the visual display of the variable's value for host-client players.
+    
+    There are two important things to keep in mind with server/host smoothing: First is that, unlike on the client side, the `OnReanticipate` callback will never be called on the server. Second is that normally updates to the value on the server side affect both `Value` and `AuthoritativeValue`, a call to `Smooth` will affect the return value of the `Value` property, but **not** the return value of the `AuthoritativeValue` property, allowing you to differentiate between the actual value (`AuthoritativeValue`) and the visual value (`Value`) that you can use for rendering the effects of the variable.
+
+- **Extrapolation:**
+
+  * *Example: A progress bar shows the time remaining to defuse a bomb. To hide latency, the progress bar starts updating the moment the player interacts with the bomb, ticks up at the same rate the server does, and finishes when it expects the server to finish. The server regularly sends updates on the state of the progress bar, so if the client and server get out of sync, the client smoothly adjusts its displayed progress bar.*
+
+  Extrapolation is a process wherein a server-controlled network variable is updated continuously to compensate for latency between the server and the client. In extrapolation, you calculate the **half**-round-trip time from the server to the client and simulate that amount of time forward for the variable, calling the same functions the server would call to calculate a new value for the variable. Generally, because server updates are not consistent in their timing, you will then also want to smooth the result as mentioned above. One thing to keep in mind, however, is that smoothing adds additional latency (as it takes that much additional time to reach the actual server value), so you will want to factor the amount of time you use for your smoothing operation into your simulation - making it half-round-trip plus smooth time.
```

Comment:
This definition of extrapolation is different from the one used in the lag page (and also different from the one used in Netcode for Entities). There we use "extrapolation" as an "unclamped interpolation" where here you include lag to predict forward

Reply:
This is one of the reasons why single source of truth for concepts is important, so that we're at least consistent in our own terms, even if the industry isn't consistent.

Reply:
I think both of those are extrapolation. The former is "linear extrapolation", which I have never personally thought was very useful since AFAIK it almost always gets wrong results. This is a more informed extrapolation taken from knowledge of how specific actors will behave.

The implementation supports both. Maybe I should add an example for linear extrapolation as well. Or do you think I should split this into two separate sections and come up with a different name for this approach (like maybe "forward simulation")?

Reply:
The difference is taking into account lag. Extrapolation is just a "continued" interpolation when you haven't received state from the server yet. In your case, your iterating forward taking into account lag. The timelines are different. Forward simulation sounds like a good name for it, not sure I'd touch "extrapolation". Maybe "lag based dead reckoning"?

Reply:
Ah, gotcha... ok, yeah, that's different from what I had heard extrapolation defined as. (I had heard of it as "trying to predict where another player would be in a shooter to account for lag so the player doesn't miss shots they think they should have hit" and even the document I had read said "this is not very useful because it's usually wrong".) In the sense of making up for data arriving late from the server, I didn't include any support for that kind of thing here; interpolation keeps the position somewhat in the past but if the interpolation window is shorter than the time between updates it'll start to get hitchy.

Funny enough, though, when I looked up the term "dead reckoning" for networking, the first result said this:

> The basic notion of dead reckoning is agreement in advance on a set of algorithms that can be used by all player nodes to **extrapolate** the behavior of entities in the game

The industry's netcode terminology really is a mess, lol.

Dead reckoning sounds like the correct term... but part of me is inclined to say "forward simulation" because I feel like less experienced developers will understand that concept better, and because it seems less likely for someone with a different idea of what dead reckoning is to object to it.

Reply:
forward simulation sounds good yeah.
just FYI, I have this PR clarifying the N4E extrapolation [here](https://github.cds.internal.unity3d.com/unity/dots/pull/12104/files#diff-3e668504b65538739e92ee4795c6cbf3cf1b137616074628dec8e3bb476a9b71R4)